CoffeeChat/View/MessageAlertView.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  MessageAlertView.swift
+//  CoffeeChat
+//
+//  Created by Lucy Xu on 4/20/20.
+//  Copyright © 2020 cuappdev. All rights reserved.
+//
+
+import UIKit
+
+protocol MessageAlertViewDelegate: class {
+    func removeAlertView(isDismiss: Bool)
+}
+
+class MessageAlertView: UIView {
+
+    // MARK: - Private View Vars
+    private let alertImageView = UIImageView()
+    private let messageLabel = UILabel()
+    private let actionButton = UIButton()
+    private let dismissButton = UIButton()
+
+    // MARK: - Private Data Vars
+    private var actionMessage: String = ""
+    private weak var delegate: MessageAlertViewDelegate?
+    private var dismissMessage: String = ""
+    private var hasDismissOption: Bool = false
+    private var mainMessage: String = ""
+
+    init(delegate: MessageAlertViewDelegate, mainMessage: String, actionMessage: String, dismissMessage: String) {
+        super.init(frame: .zero)
+        self.delegate = delegate
+        self.mainMessage = mainMessage
+        self.actionMessage = actionMessage
+        self.dismissMessage = dismissMessage
+        self.hasDismissOption = (self.dismissMessage != "")
```

Comment:
i would recommend using `dismissMessage` rather than `self.dismissMessage` just so that it isnt dependent on the order in which we declare them. also remove parens